litmus-portal/frontend/src/views/ChaosWorkflows/BrowseCluster/index.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography,
+} from '@material-ui/core';
+import moment from 'moment';
+import React, { useState } from 'react';
+import HeaderSection from './HeaderSection';
+import useStyles from './styles';
+
+interface FilterOptions {
+  search: string;
+  status: string;
+  cluster: string;
+}
+
+interface PaginationData {
+  pageNo: number;
+  rowsPerPage: number;
+}
+
+interface SortData {
+  lastRun: { sort: boolean; ascending: boolean };
+  name: { sort: boolean; ascending: boolean };
+  noOfSteps: { sort: boolean; ascending: boolean };
+}
+
+interface DateData {
+  dateValue: string;
+  fromDate: string;
+  toDate: string;
+}
+
+const BrowseCluster = () => {
+  const classes = useStyles();
+  // States for filters
+  const [filters, setFilters] = useState<FilterOptions>({
+    search: '',
+    status: 'All',
+    cluster: 'All',
+  });
+
+  // State for pagination
+  const [paginationData, setPaginationData] = useState<PaginationData>({
+    pageNo: 0,
+    rowsPerPage: 5,
+  });
+
+  const [popAnchorEl, setPopAnchorEl] = React.useState<null | HTMLElement>(
+    null
+  );
+  const isOpen = Boolean(popAnchorEl);
+
+  const [open, setOpen] = React.useState<boolean>(false);
+
+  const handlePopOverClose = () => {
+    setPopAnchorEl(null);
+    setOpen(false);
+  };
+  const handlePopOverClick = (event: React.MouseEvent<HTMLElement>) => {
+    setPopAnchorEl(event.currentTarget);
+    setOpen(true);
+  };
+
+  // State for start date and end date
+  const [dateRange, setDateRange] = React.useState<DateData>({
+    dateValue: 'Select a period',
+    fromDate: new Date(0).toString(),
+    toDate: new Date(new Date().setHours(23, 59, 59)).toString(),
+  });
+
+  // Functions passed as props in the headerSeaction
+  const changeSearch = (
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ) => {
+    setFilters({ ...filters, search: event.target.value as string });
+    setPaginationData({ ...paginationData, pageNo: 0 });
+  };
+
+  const changeStatus = (
+    event: React.ChangeEvent<{
+      name?: string | undefined;
+      value: unknown;
+    }>
+  ) => {
+    setFilters({ ...filters, status: event.target.value as string });
+    setPaginationData({ ...paginationData, pageNo: 0 });
+  };
+
+  const changeCluster = (
+    event: React.ChangeEvent<{
+      name?: string | undefined;
+      value: unknown;
+    }>
+  ) => {
+    setFilters({ ...filters, cluster: event.target.value as string });
+    setPaginationData({ ...paginationData, pageNo: 0 });
+  };
```

Comment:
Are we having pagination? if not then remove the paginationData state

Reply:
@Saranya-jena I fixed all issues. Some of the text let it be hard written anyway that is going to change.